substrate/frame/support/src/traits/tokens/fungible/mod.rs

```diff
@@ -73,6 +73,45 @@ use crate::{
 	traits::{Consideration, Footprint},
 };
 
+/// Extension for the `Consideration` trait to migrate legacy `Currency` deposits.
+///
+/// Provides a `new_from_exact` method for those types using a `fungible` balance frozen,
+/// This method is useful when a new ticket needs to be created with a precise balance, instead of
+/// deriving it from a footprint.
+pub trait FreezeConsiderationFromLegacy<A, F>: Consideration<A>
+where
+	A: 'static,
+	F: 'static + MutateFreeze<A>,
+{
+	/// Create a ticket for a `new` balance attributable to `who`. This ticket *must* ultimately
+	/// be consumed through `update` or `drop` once a footprint changes or is removed.
+	fn new_from_exact(_who: &A, _new: F::Balance) -> Result<Option<Self>, DispatchError>
```

Comment:
I believe I implemented that to return `None` as the default impl. As mentioned earlier, `ConsiderationFromLegacy` will be required for those pallets that need it during the migration.

Reply:
Hmm I'm still not convinced

1. this feels like a hack to pass a type indirectly to a migration.. As discussed in dm, I'd like to check if it's possible to pass this type directly to the migration if indeed it is only needed in the migration, not the pallet
2. there are genuine use cases for when you do want to use this directly in the pallet, e.g. @gupnik is working on a lazy migration right now where we're discussing potentially using it

Reply:
I guess the default impl would never be used and is only meant to make it compile once all currency related crates were removed from the runtime config.  
But then it should indeed return an Error and not `None`.

Reply:
Yeah would feel much better returning an error to me instead of the `None` 👍

Reply:
I still would like to see if possible to avoid the default impl (and polluting the pallet config) by passing this type directly to the migration it is used in tho.

Reply:
@liamaharon @ggwpez I replaced `None` with `Err`. It should be now good to go.

For reference, here an example of the new method being used: https://github.com/paritytech/polkadot-sdk/pull/1789/files#diff-489bbb1c7c1b1b8803ab344f63079bdc1b7d67db3d2d8809d6a3a755443619f0R77-R107